gadget/params.c

```diff
@@ -419,6 +423,8 @@ void read_parameter_file(char *fname)
         All.IO.BytesPerFile = param_get_int(ps, "BytesPerFile");
         All.IO.UsePeculiarVelocity = 0; /* Will be set by the Initial Condition File */
         All.IO.NumWriters = param_get_int(ps, "NumWriters");
+        if(All.IO.NumWriters == 0)
```

Comment:
We shall add an `unset` state to the parameters. not urgent, but I will put this on my todo list.